dns/config.go

```diff
@@ -39,6 +43,8 @@ func (c *Config) Client() (interface{}, error) {
 		return nil, fmt.Errorf("Error configuring provider: when using authentication, \"key_name\", \"key_secret\" and \"key_algorithm\" should be non empty")
 	}
 	client.c = new(dns.Client)
+	client.c.Net = c.protocol
+	client.c.Timeout = time.Duration(c.timeout) * time.Second
```

Comment:
What will happen when 0 is passed into the client here?

Reply:
Hello @katbyte,
Following source code : <https://github.com/miekg/dns/blob/master/client.go>
```
        // Timeout is a cumulative timeout for dial, write and read, defaults to 0 (disabled) - overrides DialTimeout, ReadTimeout,
	// WriteTimeout when non-zero. Can be overridden with net.Dialer.Timeout (see Client.ExchangeWithDialer and
	// Client.Dialer) or context.Context.Deadline (see the deprecated ExchangeContext)
	Timeout        time.Duration
	DialTimeout    time.Duration     // net.DialTimeout, defaults to 2 seconds, or net.Dialer.Timeout if expiring earlier - overridden by Timeout when that value is non-zero
	ReadTimeout    time.Duration     // net.Conn.SetReadTimeout value for connections, defaults to 2 seconds - overridden by Timeout when that value is non-zero
	WriteTimeout   time.Duration     // net.Conn.SetWriteTimeout value for connections, defaults to 2 seconds - overridden by Timeout when that value is non-zero
```

Reply:
Awesome, thanks!